docs/src/main/tut/datatypes/eithert.md

```diff
@@ -0,0 +1,162 @@
+resulting `EitherT` values are then composed using combinators. For example, the
+asynchronous division program can be rewritten as follows:
+
+```tut:silent
+import cats.data.EitherT
+import cats.implicits._
+
+def divisionProgramAsync(inputA: String, inputB: String): EitherT[Future, String, Double] =
+  for {
+    a <- EitherT(parseDoubleAsync(inputA))
+    b <- EitherT(parseDoubleAsync(inputB))
+    result <- EitherT(divideAsync(a, b))
+  } yield result
+
+divisionProgramAsync("4", "2").value // Future(Right(2.0))
+divisionProgramAsync("a", "b").value // Future(Left("a is not a number"))
+```
+
+Note that since `EitherT` is a monad, monadic combinators such as `flatMap` can
+be used to compose `EitherT` values.
+
+## From `A` or `B` to `EitherT[F, A, B]`
+
+To obtain a left version or a right version of `EitherT` when given an `A` or a
+`B`, use `EitherT.leftT` and `EitherT.rightT` (which is an alias for
+`EitherT.pure`), respectively.
+
+```tut:silent
+val number: EitherT[Option, String, Int] = EitherT.rightT(5)
+val error: EitherT[Option, String, Int] = EitherT.leftT("Not a number")
+```
+
+## From `F[A]` or `F[B]` to `EitherT[F, A, B]`
+
+Similary, use `EitherT.left` and `EitherT.right` to convert a `F[A]` or a `F[B]`
+into an `EitherT`. It is also possible to use `EitherT.liftT` as an alias for
+`EitherT.right`.
+
+```tut:silent
+val numberO: Option[Int] = Some(5)
+val errorO: Option[String] = Some("Not a number")
+
+val number: EitherT[Option, String, Int] = EitherT.right(numberO)
+val error: EitherT[Option, String, Int] = EitherT.left(errorO)
+```
+
+## From `Either[A, B]` or `F[Either[A, B]]` to `EitherT[F, A, B]`
+
+Use `EitherT.fromEither` to a lift a value of `Either[A, B]` into `EitherT[F, A, B]`.
+A `F[Either[A, B]]` can be converted into `EitherT` using the `EitherT` constructor.
+
+```tut:silent
+val numberE: Either[String, Int] = Right(100)
+val errorE: Either[String, Int] = Left("Not a number")
+val numberFE: List[Either[String, Int]] = List(Right(250))
+
+val numberET: EitherT[List, String, Int] = EitherT.fromEither(numberE)
+val errorET: EitherT[List, String, Int] = EitherT.fromEither(errorE)
+val numberFET: EitherT[List, String, Int] = EitherT(numberFE)
+```
+
+## From `Option[B]` or `F[Option[B]]` to `EitherT[F, A, B]`
+
+An `Option[B]` or a `F[Option[B]]`, along with a default value, can be passed to
+`EitherT.fromOption` and `EitherT.fromOptionF`, respectively, to produce an
+`EitherT`.
+
+```tut:book
+val myOption: Option[Int] = None
+val myOptionList: List[Option[Int]] = List(None, Some(2), Some(3), None, Some(5))
+
+val myOptionET = EitherT.fromOption[Future](myOption, "option not defined")
+val myOptionListET = EitherT.fromOptionF(myOptionList, "option not defined")
+```
+
+## Extracting an `F[Either[A, B]]` from an `EitherT[F, A, B]`
+
+Use the `value` method defined on `EitherT` to retrieve the underlying `F[Either[A, B]]`:
+
+```tut:book
+val errorT: EitherT[Option, String, Int] = EitherT.leftT("foo")
```

Comment:
I know it's just an example, but an `EitherT[Option, String, Int]` doesn't really seem that useful, and I'm not sure if we should include those in the official docs. Maybe change it to `EitherT[Future, String, Int]` instead? :)

Reply:
Yeah, that's much better actually.